<a href="https://colab.research.google.com/github/jabhiro/Pemograman-Dasar/blob/main/ATS_Praktikum%20Kelompok%206.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [19]:
# SOAL 1 - Percabangan / Branching
# Pemantauan suhu air pendingin sistem pembangkit
# Meminta input dari pengguna
try:
    suhu_air = float(input("Input suhu air pendingin (°C): "))

    # Mengevaluasi kondisi suhu
    if suhu_air > 100:
        status = "ALERT: SHUTDOWN"
    elif 90 < suhu_air <= 100:
        status = "STATUS: Warning"
    else:
        status = "STATUS: Normal"

    # Menampilkan hasil
    print(status)

except ValueError:
    print("Input tidak valid! Masukkan angka dengan format yang benar.")


Input suhu air pendingin (°C): 97
STATUS: Warning


Program ini menggunakan konsep percabangan (branching) dalam Python untuk mengsimulasikan pemantauan suhu air pendingin pada sistem pembangkit listrik. Ini dimulai dengan meminta pengguna memasukkan nilai suhu air pendingin dalam satuan derajat Celsius melalui fungsi input(). Nilai ini kemudian diubah menjadi tipe data float sehingga dapat digunakan dalam perhitungan numerik.  Selain itu, program menggunakan struktur percabangan if, elif, dan else untuk mengevaluasi kondisi suhu.  Jika suhu melebihi 100°C, sistem akan menampilkan pesan "ALERT: SHUTDOWN", yang menandakan kondisi berbahaya yang memerlukan penghentian sistem. Jika suhu antara 90 dan 100°C, sistem menampilkan "STATUS: Warning", yang menandakan suhu mulai tinggi dan perlu diwaspadai. Jika suhu kurang dari 90°C, sistem menampilkan "STATUS: Normal", yang menandakan kondisi aman.

Selain itu, program ini menggunakan blok try-except untuk menangani kesalahan input. Misalnya, pesan "Input tidak valid! Masukkan angka dengan format yang benar" akan muncul jika pengguna memasukkan huruf atau simbol yang bukan angka.  Ketika pengguna memasukkan suhu 97°C, program menampilkan "STATUS: Warning".  Program ini menyelesaikan masalah dengan menyediakan sistem pemantauan otomatis terhadap suhu air pendingin. Ini memungkinkan operator pembangkit untuk mengidentifikasi situasi berbahaya dengan cepat dan menghentikan kerusakan atau kegagalan sistem.

In [23]:
# Soal 3 - Looping for
# Data vibrasi dari 5 unit mesin
vibrasi_mesin = [1.2, 0.9, 2.5, 3.1, 1.8]

# Loop menggunakan enumerate untuk mendapatkan nomor unit dan nilai vibrasi
for nomor_unit, nilai_vibrasi in enumerate(vibrasi_mesin, start=1):
    if nilai_vibrasi > 2.0:
        print(f"Warning: Unit {nomor_unit} vibrasi tinggi!")


Program ini simulasi memantau tingkat getaran, atau vibrasi, dari berbagai unit mesin di sebuah sistem pembangkit.  Data getaran disimpan dalam list yang disebut vibrasi_mesin. List ini memiliki lima nilai, masing-masing menunjukkan satu unit mesin.  Untuk mendapatkan dua informasi sekaligus, nomor unit mesin dan nilai getarannya, program dapat menggunakan perulangan for bersamaan dengan fungsi enumerate().  Dengan menambahkan parameter start=1, penomoran unit dimulai dari 1, bukan 0. Selama perulangan, program menggunakan percabangan if untuk memeriksa setiap nilai getaran. Jika nilai getaran lebih besar dari 2.0, program menampilkan pesan peringatan (warning) dengan format Warning: Unit X vibrasi tinggi!, di mana X menunjukkan nomor unit yang terdeteksi memiliki getaran tinggi.

Nilai getaran unit ketiga 2,5 dan unit keempat 3,1 melebihi ambang batas 2.0, menurut data yang diberikan.  Akibatnya, dua baris peringatan ditampilkan di output program: “Warning: Unit 3 vibrasi tinggi!”  dan "Pemberitahuan: Unit 4 mengalami vibrasi tinggi!"  Mengatasi masalah pemantauan kondisi mesin, program ini membantu mendeteksi getaran berlebihan lebih awal sehingga pemeriksaan atau perawatan dapat dilakukan sebelum kerusakan menjadi lebih serius.

In [25]:
# Soal 5 - Looping while
# Sistem Pemanas - Simulasi Kenaikan Suhu

# Inisialisasi suhu awal
suhu_saat_ini = 60

# Looping selama suhu masih di bawah target
while suhu_saat_ini < 85:
    print(f"Suhu saat ini: {suhu_saat_ini}°C")
    suhu_saat_ini += 5  # Naikkan suhu sebesar 5°C setiap iterasi

# Setelah loop selesai
print("Suhu target 85°C telah tercapai!")

Suhu saat ini: 60°C
Suhu saat ini: 65°C
Suhu saat ini: 70°C
Suhu saat ini: 75°C
Suhu saat ini: 80°C
Suhu target 85°C telah tercapai!


Program ini mensimulasikan proses peningkatan suhu pada sistem pemanas. Dimulai dari suhu awal 60°C, perulangan while akan terus berjalan selama nilai suhu_saat_ini masih kurang dari 85°C. Di setiap iterasi, program mencetak suhu saat ini dan menambahkannya sebesar 5°C menggunakan operator +=. Ketika suhu mencapai atau melebihi 85°C, perulangan berhenti dan program menampilkan pesan "Suhu target 85°C telah tercapai!".

Program ini menggambarkan proses kontrol bertahap pada sistem pemanas, yaitu bagaimana suhu dinaikkan secara teratur hingga mencapai batas target yang diinginkan, sehingga dapat digunakan untuk simulasi sistem kontrol suhu dalam pembangkit listrik atau perangkat pemanas otomatis.


In [13]:
# Soal 7 - Percabangan / Branching
# Klasifikasi beban listrik berdasarkan persentase pemakaian

print("=== Sistem Monitoring Beban Listrik ===")

try:
    # Input data dari pengguna
    kapasitas_maks = float(input("Kapasitas maksimum (kW): "))
    beban_terpakai = float(input("Beban terpakai (kW): "))

    # Validasi agar kapasitas tidak nol
    if kapasitas_maks <= 0:
        print("Error: Kapasitas maksimum harus lebih dari 0.")
    else:
        # Hitung persentase pemakaian
        persen = (beban_terpakai / kapasitas_maks) * 100

        # Tentukan status beban dengan percabangan
        if persen <= 60:
            status = "Beban Ringan"
        elif persen <= 85:
            status = "Beban Sedang"
        else:
            status = "Beban Berat ⚠️"

        # Tampilkan hasil
        print(f"\nPersentase Pemakaian: {persen:.1f}%")
        print(f"Status Beban: {status}")

except ValueError:
    print("Input tidak valid! Harap masukkan angka yang benar.")


=== Sistem Monitoring Beban Listrik ===
Kapasitas maksimum (kW): 1250
Beban terpakai (kW): 984

Persentase Pemakaian: 78.7%
Status Beban: Beban Sedang


Sistem pemantauan beban listrik, seperti yang ditunjukkan pada gambar, mengklasifikasikan tingkat pemakaian daya berdasarkan persentase penggunaan terhadap kapasitas maksimum.  Program dimulai dengan menampilkan nama sistem. Selanjutnya, pengguna diminta untuk mengisi dua data penting: kapasitas maksimum (dalam kW) dan beban terpakai (dalam kW).  Agar perhitungan numerik dapat dilakukan, nilai input diubah ke tipe data float.  Selain itu, validasi dilakukan untuk memastikan kapasitas maksimum tidak negatif atau nol. Hal ini tidak logis untuk sistem listrik.  Program menghitung persentase beban yang digunakan dengan rumus (beban_terpakai/kapasitas_maks) * 100 jika data valid.  Berdasarkan hasil persentase tersebut, sistem menggunakan percabangan if-elif-else untuk menentukan status beban:

Jika pemakaian ≤ 60%, statusnya adalah "Beban Ringan",

Jika ≤ 85%, statusnya "Beban Sedang",

Dan jika > 85%, statusnya "Beban Berat ⚠️" yang menunjukkan kondisi mendekati batas maksimal.

Setelah itu, hasil perhitungan ditampilkan dalam format persentase satu desimal beserta status beban. Program juga dilengkapi dengan penanganan kesalahan (try-except) untuk menghindari error jika pengguna memasukkan data non-numerik. Pada contoh hasil yang ditunjukkan, dengan kapasitas maksimum 1250 kW dan beban terpakai 984 kW, diperoleh persentase pemakaian 78.7%, sehingga status beban adalah "Beban Sedang".

Masalah yang diselesaikan oleh program ini adalah membantu operator sistem pembangkit atau distribusi listrik dalam memantau dan mengklasifikasikan tingkat beban listrik secara otomatis, sehingga potensi kelebihan beban (overload) dapat diidentifikasi lebih awal dan sistem dapat dioperasikan dengan aman dan efisien.

In [21]:
# Soal 9 - Looping + List
# Pembacaan suhu pada 5 titik di boiler
suhu_boiler = [92.5, 88.7, 95.1, 102.3, 97.8]

print("=== Monitoring Suhu Boiler ===")

# Loop menggunakan for biasa
for indeks in range(len(suhu_boiler)):
    suhu = suhu_boiler[indeks]
    print(f"Titik {indeks + 1}: {suhu}°C")

    # Peringatan jika suhu melebihi 100°C
    if suhu > 100:
        print("ALERT: Suhu terlalu tinggi! 🔥")
    else:
        print("Status: Aman ✅")

    print("-" * 30)


=== Monitoring Suhu Boiler ===
Titik 1: 92.5°C
Status: Aman ✅
------------------------------
Titik 2: 88.7°C
Status: Aman ✅
------------------------------
Titik 3: 95.1°C
Status: Aman ✅
------------------------------
Titik 4: 102.3°C
ALERT: Suhu terlalu tinggi! 🔥
------------------------------
Titik 5: 97.8°C
Status: Aman ✅
------------------------------


Program tersebut berfungsi sebagai sistem monitoring suhu boiler yang membaca dan menampilkan kondisi suhu pada beberapa titik pengukuran di dalam sistem pembangkit. Data suhu disimpan dalam sebuah list bernama suhu_boiler, yang berisi lima nilai suhu dalam satuan derajat Celsius. Program dimulai dengan mencetak judul tampilan, kemudian melakukan perulangan menggunakan for dan fungsi range(len(suhu_boiler)) untuk mengakses setiap elemen dalam list berdasarkan indeksnya. Di setiap iterasi, program mencetak nomor titik dan nilai suhu yang terukur. Setelah itu, dilakukan pemeriksaan kondisi suhu menggunakan struktur percabangan if-else. Jika nilai suhu lebih dari 100°C, program menampilkan pesan peringatan "ALERT: Suhu terlalu tinggi! 🔥", sedangkan jika suhu berada di bawah atau sama dengan 100°C, akan ditampilkan pesan "Status: Aman ✅". Setiap hasil pengukuran dipisahkan dengan garis pembatas agar lebih mudah dibaca.

Hasil dari program menunjukkan status tiap titik, misalnya titik 1 hingga 3 berada dalam kondisi aman karena suhunya di bawah 100°C, sedangkan titik 4 memiliki suhu 102.3°C sehingga muncul peringatan "ALERT: Suhu terlalu tinggi!", dan titik 5 kembali aman.

Masalah yang diselesaikan oleh program ini adalah membantu operator dalam memantau kondisi suhu pada beberapa titik di sistem boiler secara otomatis. Dengan adanya deteksi dini terhadap titik yang mengalami kenaikan suhu berlebih, operator dapat segera melakukan tindakan preventif untuk mencegah overheating atau kerusakan pada sistem, sehingga keselamatan dan efisiensi operasi dapat terjaga.